tests/e2e/specs/miscellaneous/WorkpaseUsingParent.spec.ts

```diff
@@ -0,0 +1,99 @@
+		await browserTabsUtil.navigateTo(factoryUrl);
+		await workspaceHandlingTests.obtainWorkspaceNameFromStartingPage();
+		registerRunningWorkspace(WorkspaceHandlingTests.getWorkspaceName());
+		await projectAndFileTests.waitWorkspaceReadinessForCheCodeEditor();
+		// sometimes the trust dialog does not appear at first time, for avoiding this problem we send click event for activating
+		await new Workbench().click();
+		await projectAndFileTests.performTrustAuthorDialog();
+	});
+
+	test('Check cloning of the test project', async function (): Promise<void> {
+		const expectedProjectItems: string[] = ['.devfile.yaml', 'parent.yaml', 'README.md'];
+		const visibleContent: ViewSection = await new SideBarView().getContent().getSection('parentdevfile');
+
+		for (const expectedProjectItem of expectedProjectItems) {
+			const visibleItem: ViewItem | undefined = await visibleContent.findItem(expectedProjectItem);
+			expect(visibleItem).not.equal(undefined);
+		}
+	});
+
+	test('Check devfile VS Code tasks ', async function (): Promise<void> {
+		const input: QuickOpenBox | InputBox = await new Workbench().openCommandPrompt();
+		await input.setText('>Tasks: Run Task');
+		const runTaskItem: QuickPickItem | undefined = await input.findQuickPick('Tasks: Run Task');
+		await runTaskItem?.click();
+		const devFileTask: QuickPickItem | undefined = await input.findQuickPick('devfile');
+		await devFileTask?.click();
+		const firstExpectedQuickPick: QuickPickItem | undefined = await input.findQuickPick('1. This command from the devfile');
+		const secondExpectedQuickPick: QuickPickItem | undefined = await input.findQuickPick('2. This command from the parent');
+		expect(firstExpectedQuickPick).not.eqls(undefined);
+		expect(secondExpectedQuickPick).not.eqls(undefined);
+	});
+
+	test('Check expected containers in the parent POD', function (): void {
+		const getPodNameCommand: string = `${API_TEST_CONSTANTS.TS_API_TEST_KUBERNETES_COMMAND_LINE_TOOL} get pods --selector=controller.devfile.io/devworkspace_name=sample-using-parent --output jsonpath=\'{.items[0].metadata.name}\'`;
+
+		podName = executeArbitraryShellScript(getPodNameCommand);
+		const containerNames: string = executeArbitraryShellScript(
+			`${API_TEST_CONSTANTS.TS_API_TEST_KUBERNETES_COMMAND_LINE_TOOL} get pod ${podName} --output jsonpath=\'{.spec.containers[*].name}\'`
+		);
+		expect(containerNames).contain('tools');
+		expect(containerNames).contains('che-gateway');
+
+		const initContainerName: string = executeArbitraryShellScript(
+			`${API_TEST_CONSTANTS.TS_API_TEST_KUBERNETES_COMMAND_LINE_TOOL} get pod ${podName} --output jsonpath=\'{.spec.initContainers[].name}\'`
+		);
+		expect(initContainerName).contain('che-code-injector');
+	});
+
+	test('Check expected environment variables', function (): void {
+		const envList: string = executeArbitraryShellScript(
+			`${API_TEST_CONSTANTS.TS_API_TEST_KUBERNETES_COMMAND_LINE_TOOL} exec -i ${podName} -c tools -- sh -c env`
```

Comment:
Shell we add it into KubernetesCommandLineToolsExecutor? Like getContainerEnvs()

Reply:
Usually, we do not use it in other tests. It is specific for this functionality. But if it repeats in other test parts - of course, it is better to extract it. My point is to avoid methods that will be used for one test.  In this case, we avoid overloading `KubernetesCommandLineToolsExecutor` class.